app/uk/gov/hmrc/gform/sharedmodel/VariadicFormData.scala

```diff
@@ -188,10 +189,12 @@ case class VariadicFormData[S <: SourceOrigin](data: Map[ModelComponentId, Varia
       .distinct
       .sortBy(_.indexedComponentId.maybeIndex)
 
-  def keySet(): Set[ModelComponentId] = data.keySet
+  def keySet(): collection.Set[ModelComponentId] = data.keySet
 
   def ++[R <: SourceOrigin](addend: VariadicFormData[R]): VariadicFormData[R] = VariadicFormData[R](data ++ addend.data)
-  def addValue(entry: (ModelComponentId, VariadicValue)): VariadicFormData[S] = VariadicFormData[S](data + entry)
+  def addValue(entry: (ModelComponentId, VariadicValue)): VariadicFormData[S] = VariadicFormData[S](
+    data.concat(collection.Map.newBuilder.addOne((entry._1, entry._2)).result())
+  )
```

Comment:
```suggestion
def addValue(entry: (ModelComponentId, VariadicValue)): VariadicFormData[S] =
    fold { mutableMap =>
      mutableMap.addOne(entry)
    } { immutableMap =>
      immutableMap + entry
    }
```

Given previous suggestion `addValue` can be implemented in terms of `fold`

Reply:
Done